utils/utils.ts

```diff
@@ -1,5 +1,27 @@
+import { Lsp8AssetType } from '../contexts/AssetsContext';
+import { LSP4Metadata } from '../interfaces/lsps';
+
 export const firstTwoBytesOfAddress = (address: string) => {
   if (address) {
     return address.slice(2, 6);
   }
 };
+
+export const formatIntoLSP8AssetType = (
+  collectionName: string,
+  collectionLSP4Metadata: LSP4Metadata,
+  collectionAddress: string,
+): Lsp8AssetType => {
+  const lsp8AssetObject: Lsp8AssetType = {
+    icon: collectionLSP4Metadata.LSP4Metadata.icons[0]?.url,
+    image: collectionLSP4Metadata.LSP4Metadata.images[0][0]?.url,
+    tokenId: '',
+    description: collectionLSP4Metadata.LSP4Metadata.description,
+    collectionName: collectionName,
+    collectionDescription: collectionLSP4Metadata.LSP4Metadata.description,
+    collectionImage: collectionLSP4Metadata.LSP4Metadata.icons[0]?.url,
+    collectionIcon: collectionLSP4Metadata.LSP4Metadata.images[0][0]?.url,
+    collectionAddress: collectionAddress,
+  };
```

Comment:
you will very likely need to do try/catch here.

It will fail as sometimes `LSP4Metadata` won't exist so you will get "can get `icons` of `null`.

So maybe add more `?` and a try catch. This can break in many ways I think.

Reply:
actually I fetch and validate the `LSP4Metada` before passing it to this function 
https://github.com/lukso-network/wallet.universalprofile.cloud/blob/f7cc12673805fef12c0ac9a8e2d6f83bf6d5e39a/utils/fetchLSP4Metadata.ts#L37-L39

Reply:
amazing!